pkgs/development/python-modules/kikit/solidpython/default.nix

```diff
@@ -0,0 +1,85 @@
+# SolidPython is an unmaintained library with old dependencies.
+{ buildPythonPackage
+, callPackage
+, fetchFromGitHub
+, fetchFromGitLab
+, fetchpatch
+, lib
+, pythonRelaxDepsHook
+
+, poetry-core
+, prettytable
+, pypng
+, ply
+, setuptools
+, euclid3
+}:
+let
+  oldPypng = pypng.overrideAttrs (old: rec {
+    version = "0.0.19";
+    src = fetchFromGitLab {
+      owner = "drj11";
+      repo = "pypng";
+      rev = "refs/tags/${old.pname}-${version}";
+      hash = "sha256-XVsXgvLVFfxrRXDwdZO7oi7LPozN2XiYeXCK9NTx4Qs=";
+    };
+
+    patches = [
+      (fetchpatch {
+        url = "https://gitlab.com/drj11/pypng/-/commit/fe9c973c5e92f24746dfa1be8796c14a2befec4f.diff";
+        hash = "sha256-OYO8TE4MRHhRmgLjetk4TajFT7INQ9WdjCRjDsGc+pg=";
+      })
+    ];
+
+    disabledTests = [
+      "test_test_dir" # integration test that uses docker
+    ];
+  });
+in
+buildPythonPackage rec {
+  pname = "solidpython";
+  version = "1.1.3";
+  format = "pyproject";
+
+  src = fetchFromGitHub {
+    owner = "SolidCode";
+    repo = "SolidPython";
+    rev = "d962740d600c5dfd69458c4559fc416b9beab575";
+    hash = "sha256-3fJta2a5c8hV9FPwKn5pj01aBtsCGSRCz3vvxR/5n0Q=";
+  };
+
+  nativeBuildInputs = [ pythonRelaxDepsHook ];
+
+  propagatedBuildInputs = [
+    ply
+    setuptools
+    euclid3
+
+    prettytable
+    oldPypng
```

Comment:
We cannot do this in pythonPackages because it will cause collisions

Reply:
@SuperSandro2000, I'm not sure I understand your comment. We did *not* put `oldPypng` (or `solidPython`) in `pythonPackages`. Is the concern here about transitive dependencies (that is, someone using `kikit` (which depends on `solidPython` and this `oldPypng`) could run into issues if they also try to use a newer version of `pypng` as well)?

That does seem problematic, but is there anything we can do about that? In practice, we mostly expect people to use this kikit library inside of the kicad project, so the risk of this conflict may be low.

Reply:
hmmm, I see. Then kitkit cannot live like this in pythonPackages and should be moved somewhere close to kicat. We don't have package sets in pythonPackages.

Reply:
We removed the module from the python package set and packaged the CLI application.